### PR TITLE
Add elementsOf(Iterable) to CharSequence.contains assertion

### LINKED ISSUE #130: elementsOf(Iterable) for CharSequence.contains
*Platform* (jvm, jdk8, js, android): all

## Code related feature
```
val iterable : Iterable<Any> = listOf('h', "ello", 2019)
expect("hello world 2019").contains.exactly(1).elementsOf(iterable)

//instead of

expect("hello world 2019").contains.exactly(1).elementsOf(iterable.first(), *iterable.drop(1).toTypedArray())
```

Following the things you need to do  (see also a0b539c0 for a guideline):

*api*
- add fun `elementsOf` in apis/fluent-en_GB/atrium-fluent_en_GB-common/../charSequenceContainsCreators.kt (see other functions there)
  - it is enough if you delegate to `values`
  - `require` that the `Iterable.hasNext`
  - and `@since 0.9.0` (adopt to current version) to the KDOC
- extend CharSequenceContainsAtLeastAssertionsSpec (see TODO)

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime,
  we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](http://slack.kotlinlang.org/) in case you do not have an account yet).

### PR DESCRIPTION
**Pull request by mattyway**
resolves #130 

----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by robstoll** [comment]
@mattyway LGTM, I am not merging as long as you mark it as WIP. Do you have a particular question?

**Comment by mattyway** [comment]
Hey @robstoll, if it looks good to you I'm okay to remove the WIP tag. I was just worried about missing the coverage target.

**Comment by robstoll** [comment]
@mattyway well, it would be better if you added specs which check for empty Iterables. If you provide those, then it would look not only good but great to me :grin:

**Comment by mattyway** [comment]
Haha okay I can do that. Can you point to a file where it would be best to add those specs?

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt [hunk 1]
```diff
@@ -15,17 +15,16 @@ class CharSequenceContainsAtLeastAssertionsSpec : Spek({
...
+    include(object : ch.tutteli.atrium.specs.integration.CharSequenceContainsAtLeastAssertionsSpec(
+        getAtLeastElementsOfTriple(),
+        getAtLeastIgnoringCaseElementsOfTriple(),
+        getAtLeastButAtMostElementsOfTriple(),
+        getAtLeastButAtMostIgnoringCaseElementsOfTriple(),
+        getContainsNotPair(),
+        getExactlyPair(),
+        CharSequenceContainsAtLeastAssertionsSpec.Companion::getErrorMsgAtLeastButAtMost,
+        "◆ ", "⚬ "
+    ) {})
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceContainsAtLeastAssertionsSpec.kt]: add the test case for empty Iterables here. Something along the line of
```
include(object: Spek({
  describe("elementsOf") {  
    it("passing an empty iterable throws an IllegalArgumentException") {
      ...
    }
  }
  describe("elementsOf ignoring case"){ ... }
}) {})
```

**Comment by mattyway** [comment]
Hey @robstoll, I have added the tests for an empty iterable and the coverage check passes. Can you review again please?

**Comment by robstoll** [comment]
@mattyway thanks for the contribution :+1:
